File: Animal Age/UISnapNav.m
//
//  UISnapNav.m
//  Animal Age Converter
//
//  Created by Jon on 3/8/17.
//  Copyright © 2017 Jon Brown. All rights reserved.
//

#import "UISnapNav.h"

@interface UISnapNav ()

@end

@implementation UISnapNav

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.navigationbar.delegate = self;
    
    [[UINavigationBar appearance] setBackgroundImage:[[UIImage alloc] init]
                                      forBarPosition:UIBarPositionAny
                                          barMetrics:UIBarMetricsDefault];
    
    [[UINavigationBar appearance] setShadowImage:[[UIImage alloc] init]];
    
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (UIBarPosition)positionForBar:(id<UIBarPositioning>)bar {
    return UIBarPositionTopAttached;
}

@end

File: Animal Age/UISnapNav.swift
//
//  UISnapNav.swift
//  Animal Age
//
//  Created by Jon Brown on 12/1/19.
//  Copyright © 2019 Jon Brown. All rights reserved.
//

import Foundation
import UIKit

class UISnapNav: UINavigationController, UINavigationBarDelegate {
    @IBOutlet weak var navigationbar: UINavigationBar!

    
    override func viewDidLoad() {
        super.viewDidLoad()
        navigationbar.delegate = self
        UINavigationBar.appearance().setBackgroundImage(UIImage(), for: .any, barMetrics: .default)
        UINavigationBar.appearance().shadowImage = UIImage()
        
        
        let yellow = UIColor(red: 0.96, green: 0.84, blue: 0.09, alpha: 1.00)
       
        
        
        if #available(iOS 15.0, *) {
                let appearance = UITabBarAppearance()
                UITabBar.appearance().scrollEdgeAppearance = appearance
            
                let appearance2 = UINavigationBarAppearance()
                appearance2.backgroundColor = yellow
            
               // navigationBar.standardAppearance = appearance2
                // navigationBar.scrollEdgeAppearance = appearance2
                navigationBar.tintColor = UIColor.white

            }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    func position(for bar: UIBarPositioning) -> UIBarPosition {
        return .topAttached
    }
}
